### PR TITLE
[TINKERPOP-3006] Enable remote connection over HTTP for gremlin python

### PR DESCRIPTION
**Pull request by xiazcy**
https://issues.apache.org/jira/browse/TINKERPOP-3006

Adding experimental feature for python to connect via HTTP. This is relying on the python translator, for which additional types were added. There shouldn't be any breaking changes.

Connecting to remote can be done via
`remote_conn = DriverRemoteConnection('http://localhost':8182/, 'g')`
and for traversals
`g = traversal().with_remote(remote_conn)`

One note, in addition to the integration test, I ran the feature test suite (radish tests) locally with the HTTP endpoint for 3.6 and all tests passed. One exception is that `g.inject(null, null)` test will fail due to combination of Groovy script issue that throws NPE with 2nd nulls and server HTTP handler hanging upon NPEs. 

VOTE +1

### CODE REVIEW

File: gremlin-python/src/main/python/tests/process/test_translator.py [hunk 1]
```diff
@@ -373,6 +374,30 @@ class SuperStr(str):
         tests.append([g.V().has("p1", None),
                      "g.V().has('p1',null)"])
 
+        # 104
+        vertex = Vertex(0, "person")
```
Comment (vkagamlyk) [inline gremlin-python/src/main/python/tests/process/test_translator.py:378]: Could you add similar test for `vertex = Vertex(0, "'person'")` to confirm correct work with strings?

**Review by vkagamlyk — COMMENTED** [review]
great job, I think this is enough for experimental implementation

VOTE +1

File: gremlin-python/src/main/python/gremlin_python/driver/aiohttp/transport.py [hunk 2]
```diff
@@ -138,3 +138,104 @@ async def async_close():
...
+        # Set all inner variables to parameters passed in.
+        self._aiohttp_kwargs = kwargs
+        self._write_timeout = write_timeout
+        self._read_timeout = read_timeout
+        if "ssl_options" in self._aiohttp_kwargs:
+            self._ssl_context = self._aiohttp_kwargs.pop("ssl_options")
+            self._enable_ssl = True
+
+        # if "ssl_options" in self._aiohttp_kwargs:
+        #     self._aiohttp_kwargs["ssl_context"] = self._aiohttp_kwargs.pop("ssl_options")
```
Comment (kenhuuu) [inline gremlin-python/src/main/python/gremlin_python/driver/aiohttp/transport.py:173]: Intentional?

File: gremlin-python/src/main/python/gremlin_python/driver/client.py [hunk 3]
```diff
@@ -45,15 +46,20 @@ def __init__(self, url, traversal_source, protocol_factory=None,
...
         self._url = url
         self._headers = headers
         self._enable_user_agent_on_connect = enable_user_agent_on_connect
         self._traversal_source = traversal_source
-        if "max_content_length" not in transport_kwargs:
+        if not self._use_http and "max_content_length" not in transport_kwargs:
             transport_kwargs["max_content_length"] = 10 * 1024 * 1024
         if message_serializer is None:
             message_serializer = serializer.GraphBinarySerializersV1()
+            # message_serializer = serializer.GraphSONMessageSerializer()
```
Comment (kenhuuu) [inline gremlin-python/src/main/python/gremlin_python/driver/client.py:62]: Intentional?

**Comment by kenhuuu** [comment]
client and protocol have very similar code in lots of instances. I would consider refactoring by either adding in another class in-between to share functionality or move that shared code into functions in the parent class.

File: gremlin-python/src/main/python/tests/process/test_translator.py (hunk 1 above)
Comment (xiazcy) [reply gremlin-python/src/main/python/tests/process/test_translator.py:378]: Hmm, ok. Is that a use case to crate a label (that's a string) with a single quote? Or it's just to test?

File: gremlin-python/src/main/python/gremlin_python/driver/aiohttp/transport.py (hunk 2 above)
Comment (xiazcy) [reply gremlin-python/src/main/python/gremlin_python/driver/aiohttp/transport.py:173]: Ah yes, I forgot to remove this part.

File: gremlin-python/src/main/python/gremlin_python/driver/client.py (hunk 3 above)
Comment (xiazcy) [reply gremlin-python/src/main/python/gremlin_python/driver/client.py:62]: Also something I forgot to remove, thanks!

File: gremlin-python/src/main/python/tests/process/test_translator.py (hunk 1 above)
Comment (vkagamlyk) [reply gremlin-python/src/main/python/tests/process/test_translator.py:378]: create label with single quote to verify serialization

**Comment by Cole-Greer** [comment]
Thanks Yang, LGTM. It's nice to see some additions to the Python translator included here as well.

VOTE +1

File: gremlin-python/src/main/python/tests/process/test_translator.py (hunk 1 above)
Comment (xiazcy) [reply gremlin-python/src/main/python/tests/process/test_translator.py:378]: Updated to using repr() for string formatting for now.

**Pull request merged**